packages/edit-site/src/components/block-editor/inserter-media-categories.js

```diff
@@ -0,0 +1,161 @@
+	if ( ! isPublicDomainMark ) {
+		licenseName = `CC ${ licenseName }`;
+	}
+	if ( !! licenseUrl ) {
+		license = getExternalLink(
+			`${ licenseUrl }?ref=openverse`,
+			licenseName
+		);
+	}
+	const markedLicence = isPublicDomainMark
+		? sprintf(
+				/* translators: %s: Name of the media's license (ex. 'CC0 1.0'). */
+				__( 'is marked with %s' ),
+				license
+		  )
+		: sprintf(
+				/* translators: %s: Name of the media's license (ex. 'CC BY-ND 2.0'). */
+				__( 'is licensed under %s' ),
+				license
+		  );
+	return markedLicence;
+};
+
+const getOpenverseCaption = ( item ) => {
+	const {
+		title,
+		foreign_landing_url: foreignLandingUrl,
+		creator,
+		creator_url: creatorUrl,
+		license,
+		license_version: licenseVersion,
+		license_url: licenseUrl,
+	} = item;
+	/* translators: Openverse default media item title in the block inserter's media list. (ex. 'This work by {creator} is marked with CC0 1.0'). */
+	const defaultTitle = __( 'This work' );
+	let _title = title ? decodeEntities( title ) : defaultTitle;
+	_title = getExternalLink( foreignLandingUrl, _title );
+	const fullLicense = getOpenverseLicense(
+		license,
+		licenseVersion,
+		licenseUrl
+	);
+	let _creator = decodeEntities( creator );
+	if ( creatorUrl ) {
+		_creator = getExternalLink( creatorUrl, creator );
+	}
+	_creator = sprintf(
+		/* translators: %s: Name of the media's creator. */
+		__( ' by %s' ),
+		_creator
+	);
+	const caption = `"${ _title }" ${ _creator } ${ fullLicense }.`;
+	return caption.replace( /\s{2}/g, ' ' );
+};
+
+const coreMediaFetch = async ( query = {} ) => {
+	const mediaItems = await resolveSelect( coreStore ).getMediaItems( {
+		...query,
+		orderBy: !! query?.search ? 'relevance' : 'date',
+	} );
+	return mediaItems.map( ( mediaItem ) => ( {
+		...mediaItem,
+		alt: mediaItem.alt_text,
+		url: mediaItem.source_url,
+		caption: mediaItem.caption?.raw,
+	} ) );
+};
+
+const inserterMediaCategories = [
+	{
+		label: __( 'Images' ),
+		name: 'images',
+		mediaType: 'image',
+		async fetch( query = {} ) {
+			return coreMediaFetch( { ...query, media_type: 'image' } );
+		},
+	},
+	{
+		label: __( 'Videos' ),
+		name: 'videos',
+		mediaType: 'video',
+		async fetch( query = {} ) {
+			return coreMediaFetch( { ...query, media_type: 'video' } );
+		},
+	},
+	{
+		label: __( 'Audio' ),
+		name: 'audio',
+		mediaType: 'audio',
+		async fetch( query = {} ) {
+			return coreMediaFetch( { ...query, media_type: 'audio' } );
+		},
+	},
+	{
+		label: 'Openverse',
+		searchLabel: 'Openverse',
+		name: 'openverse',
+		mediaType: 'image',
+		description: <p>{ __( 'Disclaimer message' ) }</p>,
+		async fetch( query = {} ) {
+			const defaultArgs = {
+				mature: false,
+				excluded_source: 'flickr',
+				license: 'cc0',
+			};
+			const finalQuery = { ...query, ...defaultArgs };
+			const mapFromInserterMediaRequest = {
+				per_page: 'page_size',
+				search: 'q',
+			};
+			const url = new URL(
```

Comment:
Do we add a user agent values to the request? It would be great to determine that the API request came from Gutenberg for analytics purposes.

Reply:
I'm not sure we can do this for privacy reasons and not sure who to ping about that.. maybe @priethor or @annezazu?

Reply:
> I'm not sure we can do this for privacy reasons

@obulat's suggestion would be that it's a generic user-agent that indicates the requests are coming from _Gutenberg_, nothing user-specific. That would definitely help Openverse track and understand how much traffic we're seeing from this integration, if it is possible and respects the privacy of users.

Reply:
To give some potentially helpful context here, the concern about privacy with a user-agent shared across all WordPress installations is not a realistic one. Fetch [already sends a referrer header by default unless told to do otherwise](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Referrer-Policy). Openverse's API will have access to the IP of the requester, the domain they're requesting from, and the query strings. It would be trivial (though tedious and relatively imprecise) to deduce which requests are coming from WordPress installations. Adding a shared user agent to the request would not de-anonymise requesters but would help Openverse monitor and understand sources of traffic and usage.

Reply:
Done here: https://github.com/WordPress/gutenberg/pull/46251/commits/9721f8f2bafa73ed82bff7675d56dc1c42596029